### DISCUSSION TITLE
How is Change Failure Rate calculated?

### DISCUSSION THREAD

**Discussion by mewid** [middlewarehq/middleware#359] [Q&A] [answered]
I have read the README describing the CFR being "computed by linking incidents to deployments within an interval; each deployment may have several or no incidents."

I feel like im lacking knowledge here of what an incident would mean in a GitHub context? The information links leads me to GitLab, which seems to have incident management, but if im not wrong, Middleware does not have GitLab integration? Either way its unclear to me how this works with GitHub integration.

**Comment by samad-yar-khan** [comment, marked as answer]
Hey @mewid, I understand the confusion. We mentioned GitLab to explain how an incident provider works.

Using GitHub, the incident related metrics are shown by considering the Reverted PRs as Incidents.

The Incident’s creation timestamp is set at the reverted PRs merge timestamp and the incident resolution time stamp is taken as the merge time of the revert PRs and the resolution time the difference between the two timestamps.

Thanks for raising the concern, please let me know if this discussion resolved your issue or if you have anymore doubts regarding the product.

We are looking forward to add more integrations like GitLab and would highly appreciate your support 🙌

**Reply by mewid** [reply]
Aha, thank you for the clarification! I see now that I didn't read this part carefully: "For simplicity, we synchronize your Pull Request data and classify reverted Pull Requests as incidents and merged Pull Requests as Deployments.".
